pandas/_libs/lib.pyi

```diff
@@ -23,15 +24,15 @@ from pandas._typing import (
 )
 
 # placeholder until we can specify np.ndarray[object, ndim=2]
-ndarray_obj_2d = np.ndarray
+ndarray_obj_2d: TypeAlias = np.ndarray
 
 from enum import Enum
 
 class _NoDefault(Enum):
     no_default = ...
 
-no_default: Final = _NoDefault.no_default
-NoDefault = Literal[_NoDefault.no_default]
+no_default: Final = ...
+NoDefault = ...
```

Comment:
These changes also caused errors for mypy&pyright: https://github.com/pandas-dev/pandas/actions/runs/4620671070/jobs/8171100725?pr=52452

If
```py
no_default: Final[TypeAlias] = _NoDefault.no_default
NoDefault: TypeAlias = Literal[_NoDefault.no_default]
```

doesn't work, you can suppress those errors.

Reply:
Might need to ignore these two lines, mypy reports many errors like these:

> error: Expression of type "TypeAlias" cannot be assigned to parameter of type "bool | Literal[_NoDefault.no_default]"
>     Type "TypeAlias" cannot be assigned to type "bool | Literal[_NoDefault.no_default]"